src/app/components/edit-nonce-button.tsx

```diff
@@ -0,0 +1,28 @@
+import { useNavigate } from 'react-router-dom';
+
+import { BoxProps, color } from '@stacks/ui';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { Caption } from '@app/components/typography';
+
+interface EditNonceButtonProps extends BoxProps {
+  params?: object;
+}
+export function EditNonceButton({ params, ...props }: EditNonceButtonProps) {
+  const navigate = useNavigate();
+
+  return (
+    <Caption
+      _hover={{ cursor: 'pointer', textDecoration: 'underline' }}
+      alignSelf="start"
+      as="button"
+      color={color('brand')}
+      onClick={() => navigate(RouteUrls.EditNonce, { state: params })}
+      type="button"
+      {...props}
+    >
+      Edit nonce
+    </Caption>
+  );
+}
```

Comment:
Tidy component, though doesn't make sense to me to pass in params to then call in navigate, instead you could emit an `onEditNonce` event? and call `navigate(RouteUrls.EditNonce, { state: params })` in parent?

Reply:
Just seems a bit redundant in each parent when the contractId is only needed with fungible tokens?